src/lib/transaction_logic/test/zkapp_logic.ml

```diff
@@ -819,5 +955,200 @@ let%test_module "Test transaction logic." =
                       ] )
                  && Predicates.verify_balances_unchanged ~ledger ~txn accounts )
             )
-            (run_zkapp_cmd ~fee_payer ~fee ~accounts txns) )
+            (run_zkapp_cmd ~global_slot ~fee_payer ~fee ~accounts txns) )
+
+    let%test_unit "Update preconditions can block update application." =
```

Comment:
Could you add comment to tests in this file on why you expect them to fail (for those that are to fail)?

P.S. I inferred it from this test, but it took a while

Reply:
Done in 091f85a7de.